src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int START_ARRAY_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private double threshold;
+    private Node<K, V> currentNode;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[START_ARRAY_CAPACITY];
+        threshold = table.length * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
```

Comment:
Your put method is way bigger than needed, try to simplify it:
like:
```
public void put(K key, V value) {
        resize();

        int index = getIndex(key);
        Node<K, V> node = table[index];
        while (node != null) {
            if (Objects.equals(node.key, key)) {
                node.value = value;
                return;
            }
            node = node.next;
        }
        Node<K, V> nodeNew = new Node<>(key, value, table[index]);
        table[index] = nodeNew;
        size++;
    }
```

Reply:
so we put new node to the 0 position? we just need them to be by that index and no matter which one of them is first?

Reply:
It doesn't matter what the order is. On get operation we go to the bucket and check keys with equals one-by-one.
In general amount of collisions shouldn't be significant so you will have 1-2 nodes in the bucket